api/v3/Order.php

```diff
@@ -59,3 +59,42 @@ function civicrm_api3_order_get($params) {
   $params['sequential'] = $isSequential;
   return civicrm_api3_create_success($contributions, $params, 'Order', 'get');
 }
+
+/**
+ * Delete a Order.
+ *
+ * @param array $params
+ *   Input parameters.
+ *
+ * @return array
+ */
+function civicrm_api3_order_delete($params) {
+  $contribution = civicrm_api3('Contribution', 'get', array(
+    'return' => array('is_test'),
+    'id' => CRM_Utils_Array::value('contribution_id', $params, $params['id']),
+  ));
+  if ($contribution['id'] && $contribution['values'][$contribution['id']]['is_test'] == TRUE) {
+    $result = civicrm_api3('Contribution', 'delete', $params);
+  }
+  else {
+    throw new API_Exception('Could not delete Order.');
+  }
+  return civicrm_api3_create_success(CRM_Utils_Array::value('values', $result), $params, 'Order', 'delete');
```

Comment:
CRM_Utils_Array::value('values', $result) shouldn't be needed - $result['values'] should always be set & if it's not that is  a problem & an e-notice is appropriate